### PR TITLE
No need to delete the merged anonymous basket after merge

### PR DESCRIPTION
**Pull request by aleksandrpanteleymonov**
For some reason the merged basket was deleted but oscar already has a status for that - MERGED. It makes super difficult to debug complex cases if anonymous basket got deleted.

### CODE REVIEW

**Comment by maerteijn** [comment]
@specialunderwear Your thoughts?

**Comment by specialunderwear** [comment]
I think that deleting the hook is wrong. It is there to implement all kinds of merge strategies and apply rules when merging. For example, it could be you are not allowed 2 products of the same kind in the basket, the merge_baskets hook is the place to implement a merge strategy to avoid a merge bringing 2 items in a basket.

The rest if fine, leave the anonymous basket and change the status to merged. You should document this in the release notes because some systems could depend on the deletion of the anonymous basket to avoid basket table blowup.

File: oscarapi/views/login.py [hunk 1]
```diff
@@ -54,11 +54,6 @@ def get(self, request, format=None):
 
         raise MethodNotAllowed('GET')
 
-    def merge_baskets(self, anonymous_basket, basket):
```
Comment (specialunderwear) [inline oscarapi/views/login.py]: don't delete the merge basket hook.

**Comment by maerteijn** [comment]
Ok I agree with keeping the hook there so you can easily implement different merging strategies. @aleksandrpanteleymonov if you adjust this I will merge it and document this in the release notes.

**Comment by aleksandrpanteleymonov** [comment]
@maerteijn I restored the hook in `LoginView`. The reason behind the initial decision to delete it was that `Basket` class already has the method `merge` for implementing different merging strategies. But I think it is fine if oscar-api has an option to redefine it.

**Comment by maerteijn** [comment]
@aleksandrpanteleymonov Yes, basket already has a `merge` method, but if you're fine with oscar's implementation you don't need to completely override the basket app and model just to adjust this.

@specialunderwear I think this pull request can be merged and will need explicitly stated release notes of the change when I will release it. Agree to merge it?

**Pull request merged**